Description: Python 3 fix for sphinx doc
 In Python 3, dict methods dict.keys() return “views” instead of lists.
 Therefore, the code:
 .
    impls = sorted(matrix.targets.keys())
    impls.sort()
 .
 should be replaced by:
 .
    impls = sorted(matrix.targets.keys())
 .
 to be Python 3 compatible.
Author: Thomas Goirand <zigo@debian.org>
Date: Thu, 15 Feb 2018 14:21:58 +0100
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Change-Id: I50477b0ac43adff9e9b0220bd7c48d6ebb6471b3
Forwarded: https://review.openstack.org/#/c/544956/
Last-Update: 2018-02-15

diff --git a/doc/ext/feature_matrix.py b/doc/ext/feature_matrix.py
index 1657750..4934e58 100644
--- a/doc/ext/feature_matrix.py
+++ b/doc/ext/feature_matrix.py
@@ -337,8 +337,7 @@ class FeatureMatrixDirective(rst.Directive):
         summaryhead.append(header)
 
         # then one column for each hypervisor driver
-        impls = matrix.targets.keys()
-        impls.sort()
+        impls = sorted(matrix.targets.keys())
         for key in impls:
             target = matrix.targets[key]
             implcol = nodes.entry()
@@ -378,8 +377,7 @@ class FeatureMatrixDirective(rst.Directive):
                 classes=["fm_maturity_" + feature.maturity]))
 
             # and then one column for each hypervisor driver
-            impls = matrix.targets.keys()
-            impls.sort()
+            impls = sorted(matrix.targets.keys())
             for key in impls:
                 target = matrix.targets[key]
                 impl = feature.implementations[key]
-- 
cgit v1.1

